packages/react/src/components/Pagination/Pagination.tsx

```diff
@@ -15,10 +15,18 @@ interface Props extends React.HTMLAttributes<HTMLDivElement> {
   previousPageLabel?: string;
   nextPageLabel?: string;
   lastPageLabel?: string;
-  onNextPageClick?: () => void;
-  onPreviousPageClick?: () => void;
-  onFirstPageClick?: () => void;
-  onLastPageClick?: () => void;
+  onNextPageClick?: (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
```

Comment:
You should just be able to use `React.MouseEvent<HTMLButtonElement>` here (and for the other functions). React defaults the [second generic] to `NativeMouseEvent` by default.

```suggestion
    event: React.MouseEvent<HTMLButtonElement>
```

Reply:
Whoops, will update